### ISSUE TITLE
Problem with adding CC licenses in DSpace 7.6

### ISSUE THREAD

**Issue by Manius71** [DSpace/DSpace#9060]
**Describe the bug**
When adding a Creative Commons license in DSpace 7.6, I receive an error message as shown in the screenshot:
![image](https://github.com/DSpace/DSpace/assets/26096252/7bf165d8-54d5-4a8b-96bd-fc73831e417b)
Additionally, it is accompanied by the following error in the DSpace logs:
2023-09-02 00:23:29,954 ERROR 873d3483-5258-42a6-be03-5414a7773662 30382c93-cefc-4775-bd1a-3ce6c8694988 org.dspace.license.CCLicenseConnectorServiceImpl @ Error while retrieving the license document for URI: http://creativecommons.org/licenses/by/3.0/us/
2023-09-02 00:23:29,963 ERROR 873d3483-5258-42a6-be03-5414a7773662 30382c93-cefc-4775-bd1a-3ce6c8694988 org.dspace.app.rest.submit.SubmissionService @ The license uri: http://creativecommons.org/licenses/by/3.0/us/, could not be resolved to a CC license
2023-09-02 00:23:29,965 ERROR 873d3483-5258-42a6-be03-5414a7773662 30382c93-cefc-4775-bd1a-3ce6c8694988 org.dspace.app.rest.exception.DSpaceApiExceptionControllerAdvice @ An exception has occurred (status:500)

**To Reproduce**
1. Go to New and click on Item.
2. After opening the item submission form, go to the Creative Commons License section.
3. Select a license type: Creative Commons.
4. After selecting the option for this license, check the box next to "I grant the license above."
5. Click Save, and an error message appears, as shown in the screenshot.

**Expected behavior**
Correct operation of the Creative Commons license selection function.

**Related work**
I couldn't find any information about someone reporting the same issue. I will add that I didn't encounter such problems in versions 7.4 and 7.5.

**Comment by tdonohue** [comment]
@Manius71 : I'm not able to reproduce this with default settings.   Here's what I've tried:

1. I'm using all the default values for `cc.*` settings as shown here: https://github.com/DSpace/DSpace/blob/main/dspace/config/dspace.cfg#L976-L1014
2. In my `item-submission.xml`, I've uncommented the 'cclicense' step here: https://github.com/DSpace/DSpace/blob/main/dspace/config/item-submission.xml#L277
3. Then, I've started a new submission.  I can see the CC license step.  I selected "Creative Commons" to add a license, and answered "Yes" to both questions and checked the checkbox.  
4. I then completed the submission with no errors.  Double checked my logs and no errors appear their either.

Is it possible you've made some other customization to one of the "cc.*" configs in your dspace.cfg or local.cfg?   I'm not sure how this is working for me and not for you.

**Comment by Manius71** [comment]
I'm inserting my settings:

local.cfg:
cc.license.classfilter = recombo, mark, publicdomain, zero
cc.license.jurisdiction = pl
cc.license.locale = pl

dspace.cfg:
cc.api.rooturl = https://api.creativecommons.org/rest/1.5
cc.license.uri = dc.rights.uri
cc.license.name = dc.rights
cc.submit.setname = true
cc.submit.addbitstream = true
cc.license.classfilter = publicdomain, recombo, mark
cc.license.jurisdiction = us
cc.license.locale = en

But as I mentioned on the discussion group, I commented out the settings in local.cfg, and after the restart, I received the messages that I pasted here.

In this situation, the question is, what else can I check?

**Comment by tdonohue** [comment]
@Manius71 : Have you tested this again recently?  It *is* entirely possible that it was something wrong in the Creative Commons API itself at the moment when you tested it.  (In other words, if the CC API goes down or is inaccessible, then DSpace's CC Step won't work.)

Or it could be an issue with your DSpace site communicating with the Creative Commons API at  https://api.creativecommons.org/rest/1.5   If your DSpace is blocking that URL or somehow firewalled, then it's possible this won't work properly.

DSpace's Creative Commons step *requires* that DSpace can access the Creative Commons API.  So, you could verify that you are able to communicate with the CC API via commandline tools like "curl" or "wget" without errors being thrown.

**Comment by Manius71** [comment]
I've tested it multiple times, including just now - the result is the same.

I've also conducted tests, and here are the results:

curl https://api.creativecommons.org/rest/1.5

[testcurl1.txt](https://github.com/DSpace/DSpace/files/12579678/testcurl1.txt)

curl http://creativecommons.org/licenses/by/3.0/us/

[testcurl2.txt](https://github.com/DSpace/DSpace/files/12579684/testcurl2.txt)

With option -I (big i)
curl -I http://creativecommons.org/licenses/by/3.0/us/

[testcurl_I.txt](https://github.com/DSpace/DSpace/files/12579698/testcurl_I.txt)

But if I used the -L option to make curl follow redirects, the result is correct:

[testcurl_L.txt](https://github.com/DSpace/DSpace/files/12579666/testcurl_L.txt)

In summary, I get the impression that the issue in DSpace may be related to the redirection from http to https. However, it works for you. Strange.

**Comment by tdonohue** [comment]
I have to admit, I'm confused as well.  Every time I try on my end, it works fine. 

Is there maybe some sort of proxy on your end?  I see this other ticket mentions some sites having trouble communicating with Creative Commons if a proxy is in place: https://github.com/DSpace/DSpace/issues/8603#issuecomment-1621209161

I'm honestly not sure what else to try.  On my end, I'm running the UI/REST API simply on localhost.  I don't believe that should make a difference. But, it's also possible something in your Apache/Nginx settings could be causing those 301 redirects to not be followed?

It seems obvious to me that there must be some sort of setup where this isn't fully working. I'm just not sure how to reproduce it on my end.  In any case, I'll leave this ticket open in the hopes that _others may be able to reproduce it_ and provide us with more clues.

**Comment by Manius71** [comment]
I will continue to analyze the issue in this direction as well. However, as I mentioned, it was working with DSpace 7.4 and 7.5. Nothing much has changed in the meantime (we are using nginx).

**Comment by tdonohue** [comment]
Yes, I understand.  It's just very odd to me because I'm only aware of *two* changes to CC Licensing that occurred in 7.6:  #8665 and #8903  Neither of those seem like they'd impact this, and obviously this continues to work on my end.  So, it's furthers the mystery.

I've asked on Slack to see if any other developers can chime in here.  I'm not understanding what could be going on & how this works fine on my end, but doesn't for you.

**Comment by Manius71** [comment]
Thank you, Tim, for your assistance so far.

**Comment by toniprieto** [comment]
@Manius71 In case it helps, I've seen in one of [your message](https://groups.google.com/g/dspace-tech/c/Tas1O_drCko/m/UNbAQ2FCAQAJ) to dspace-tech mailing list that you have sent a part of the dspace.log that includes a stack trace error with more information. The stack trace that begins with:

```
java.net.SocketException: java.security.NoSuchAlgorithmException: Error constructing implementation (algorithm: Default, provider: SunJSSE, class: sun.security.ssl.SSLContextImpl$DefaultSSLContext)
	at javax.net.ssl.DefaultSSLSocketFactory.throwException(SSLSocketFactory.java:267) ~[?:?]
	at javax.net.ssl.DefaultSSLSocketFactory.createSocket(SSLSocketFactory.java:274) ~[?:?]
	at sun.net.www.protocol.https.HttpsClient.createSocket(HttpsClient.java:422) ~[?:?]
...
```

I would search more info about this error: "java.security.NoSuchAlgorithmException: Error constructing implementation (algorithm: Default, provider: SunJSSE, class: sun.security.ssl.SSLContextImpl$DefaultSSLContext)"

For example: https://stackoverflow.com/questions/9761575/java-nosuchalgorithmexception-sunjsse-sun-security-ssl-sslcontextimpldefault

Could it be something related with the ssl/https configuration?

**Comment by AgustinAlfieri** [comment]
Hi @Manius71, I don't know if this helps but I had a similar problem reported by some users. We find out that it was caused by a collection template that had the field already completed. My best guess is that this caused Dspace to try to save 2 different values on the same dc field and that was the cause of the issue.

**Comment by Manius71** [comment]
Hello @AgustinAlfieri, I checked your suggestion, but without success. Firstly, we do indeed use collection templates, but we only use one metadata field in them, which is dc.type. Secondly, I conducted a test on a collection without any template, and the error still occurs.
So it seems like a different case. Nonetheless, thank you for your assistance.

**Comment by yinyinlatt** [comment]
Hi All

This issue appears to be related to the Java version used with DSpace. I recently encountered the similar problem after an OS auto-patching upgraded Java from version 17 to 21.

java -version : openjdk version "17.0.20.1" 2026-08-18 LTS 
Result: DSpace 8.x operates normally.

java -version : openjdk version "21.0.12.1" 2026-08-18 LTS 
Result: DSpace 8.x experiences Creative Commons license retrieval errors (see error below).

Regards,
Yin Yin 
Connect | Digital Services
The University of Auckland | Te Whare Wānanga o Tāmaki Makaurau
https://orcid.org/0000-0002-3988-6895
E: y.latt@auckland.ac.nz


2026-09-11 08:06:39,855 ERROR ba941944-b246-4542-8385-fad9e3ec6935 f71aaf57-c4ed-4791-a4b9-1f1ed164b784 org.dspace.license.CCLicenseConnectorServiceImpl @ Error while retrieving the license document for URI: http://creativecommons.org/licenses/by-nc-nd/4.0/
javax.net.ssl.SSLException: (internal_error) Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:132) ~[?:?]
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:383) ~[?:?]
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:326) ~[?:?]
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:321) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1709) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:470) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426) ~[?:?]
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436) ~[httpclient-4.5.14.jar!/:4.5.14]
        at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.upgrade(DefaultHttpClientConnectionOperator.java:191) ~

2026-09-11 08:06:39,857 ERROR ba941944-b246-4542-8385-fad9e3ec6935 f71aaf57-c4ed-4791-a4b9-1f1ed164b784 org.dspace.app.rest.submit.SubmissionService @ The license uri: http://creativecommons.org/licenses/by-nc-nd/4.0/, could not be resolved to a CC license
java.lang.IllegalArgumentException: The license uri: http://creativecommons.org/licenses/by-nc-nd/4.0/, could not be resolved to a CC license
        at org.dspace.app.rest.submit.factory.impl.CCLicenseAddPatchOperation.add(CCLicenseAddPatchOperation.java:62) ~[dspace-server-webapp-8.2.jar!/:8.2]
        at org.dspace.app.rest.submit.factory.impl.AddPatchOperation.perform(AddPatchOperation.java:26) ~[dspace-server-webapp-8.2.jar!/:8.2]
        at org.dspace.app.rest.submit.step.CCLicenseStep.doPatchProcessing(CCLicenseStep.java:61) ~[dspace-server-webapp-8.2.jar!/:8.2]
        at org.dspace.app.rest.submit.SubmissionService.evaluatePatchToInprogressSubmission(SubmissionService.java:504) ~[dspace-server-webapp-8.2.jar!/:8.2]
        at org.dspace.app.rest.repository.WorkspaceItemRestRepository.patch(WorkspaceItemRestRepository.java:213) ~[dspace-server-webapp-8.2.jar!/:8.2]
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
        at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]

**Comment by tdonohue** [comment]
Thanks for those details @yinyinlatt !  

In that case, I'm closing this ticket as "answered".  The problem appears to be that you may be using an unsupported version of Java / JDK.  DSpace 7 / 8 / 9 **all require Java 17** and will not function properly on Java 21 (see the installation guide for all those releases).  If you want to use Java 21, then you should upgrade to DSpace 10, as that's the first version that supports Java 21.
